internal/tiltfile/tiltfile_state.go

```diff
@@ -535,6 +541,35 @@ func (s *tiltfileState) assemble() (resourceSet, []k8s.K8sEntity, error) {
 	}, s.k8sUnresourced, nil
 }
 
+// Emit an error if there are unmatches images.
+//
+// There are 3 mistakes people commonly make if they
+// have unmatched images:
+// 1) They didn't include any Kubernetes or Docker Compose configs at all.
+// 2) They included Kubernetes configs, but they're custom resources
+//    and Tilt can't infer the image.
+// 3) They typo'd the image name, and need help finding the right name.
```

Comment:
The one that @jgoulah seemed to be encountering is a slight variation on this, because of misconfiguration the YAML that should have consumed this was never included.

I don't think it's possible to account for this at this point in the code, I think this is more of a job for facets. Just wanted to note it here.

Reply:
its similar to a typo but it also pops up when helm values don't render the image tag or if tht tag isn't what tilt expects

Reply:
ya, i added some info to this to the comment. I have ideas on tooling to help with this, though it's slightly different than the facets approach. stay tuned!